Userland/Applications/Presenter/Presentation.cpp

```diff
@@ -147,15 +146,45 @@ ErrorOr<Gfx::IntSize> Presentation::parse_presentation_size(JsonObject const& me
     };
 }
 
-void Presentation::paint(Gfx::Painter& painter) const
+ErrorOr<DeprecatedString> Presentation::render()
 {
-    auto display_area = painter.clip_rect();
-    // These two should be the same, but better be safe than sorry.
-    auto width_scale = static_cast<double>(display_area.width()) / static_cast<double>(m_normative_size.width());
-    auto height_scale = static_cast<double>(display_area.height()) / static_cast<double>(m_normative_size.height());
-    auto scale = Gfx::FloatSize { static_cast<float>(width_scale), static_cast<float>(height_scale) };
-
-    // FIXME: Fill the background with a color depending on the color scheme
-    painter.clear_rect(painter.clip_rect(), Color::White);
-    current_slide().paint(painter, m_current_frame_in_slide.value(), scale);
+    HTMLElement main_element;
+    main_element.tag_name = "main"sv;
+    for (size_t i = 0; i < m_slides.size(); ++i) {
+        HTMLElement slide_div;
+        slide_div.tag_name = "div"sv;
+        TRY(slide_div.style.try_set("display"sv, "none"sv));
+        TRY(slide_div.attributes.try_set("id"sv, DeprecatedString::formatted("slide{}", i)));
+        TRY(slide_div.attributes.try_set("class"sv, "slide"));
+        auto& slide = m_slides[i];
+        TRY(slide_div.children.try_append(TRY(slide.render(*this))));
+        main_element.children.append(move(slide_div));
+    }
+
+    StringBuilder builder;
+    TRY(builder.try_append(R"(
+<!DOCTYPE html><html><head><style>
+    .slide {
+        position: absolute;
+        left: 0;
+        top: 0;
+        width: 100%;
+        height: 100%;
+    }
+</style><script>
+    function goto(slideIndex, frameIndex) {
+        // FIXME: Honor the frameIndex.
+        let slide;
+        for (slide of document.getElementsByClassName("slide")) {
+            slide.style.display = "none";
+        }
+        if (slide = document.getElementById(`slide${slideIndex}`))
```

Comment:
you need at least 1 more `=` to make this do what I presume you intended

Reply:
nope:

<img width="422" alt="image" src="https://user-images.githubusercontent.com/19366641/211513738-b35025aa-16ef-4372-a169-61724934b001.png">

you can also see this working in the video :)